src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -1,19 +1,59 @@
 package core.basesyntax.impl;
 
+import core.basesyntax.Entry;
 import core.basesyntax.Storage;
 
+@SuppressWarnings("-Xlint:unchecked")
 public class StorageImpl<K, V> implements Storage<K, V> {
+
+    private static final int ENTRY_ITEMS_NUMBER = 10;
+    private final Entry<K, V>[] entries;
+    private int size;
+
+    public StorageImpl() {
+        entries = new Entry[ENTRY_ITEMS_NUMBER];
+        size = 0;
+    }
+
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < size; i++) {
+            if (isEqualKeys(key, entries[i].getKey())) {
+                entries[i].setValue(value);
+                return;
+            }
+        }
+        try {
+            entries[size] = new Entry<>(key, value);
+            size++;
+        } catch (IndexOutOfBoundsException e) {
+            throw new RuntimeException("The storage can have a maximum of"
+                    + ENTRY_ITEMS_NUMBER + " entries/", e);
+        }
```

Comment:
```suggestion
        try {
            entries[size] = new Entry<>(key, value);
            size++;
        } catch (IndexOutOfBoundsException e) {
            throw new RuntimeException("The storage can have a maximum of"
                    + ENTRY_ITEMS_NUMBER + " entries/", e);
        }
```

Not required for this assignment. But it's great that you figured it out!